spring-integration-core/src/test/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiatorTests.java

```diff
@@ -284,17 +290,41 @@ public void testExceptionFromLock() throws Exception {
 
 		CountDownLatch onGranted = new CountDownLatch(1);
 
-		LockRegistryLeaderInitiator initiator = new LockRegistryLeaderInitiator(registry);
+		LockRegistryLeaderInitiator another = new LockRegistryLeaderInitiator(registry);
 
-		initiator.setLeaderEventPublisher(new CountingPublisher(onGranted));
+		another.setLeaderEventPublisher(new CountingPublisher(onGranted));
 
-		initiator.start();
+		another.start();
 
 		assertTrue(onGranted.await(10, TimeUnit.SECONDS));
-		assertTrue(initiator.getContext().isLeader());
+		assertTrue(another.getContext().isLeader());
 		assertTrue(exceptionThrown.get());
 
-		initiator.stop();
+		another.stop();
+	}
+
+	@Test
+	public void shouldShutdownInternalExecutorService() throws Exception {
+
+		this.initiator.start();
+		this.initiator.destroy();
+
+		ExecutorService executorService = (ExecutorService) TestUtils.getPropertyValue(this.initiator, "executorService");
```

Comment:
There is an overloaded `getPropertyValue()` with an explicit class and generic argument to return.